pkg/kyverno/apply/command.go

```diff
@@ -0,0 +1,267 @@
+	var resourcePath, kubeConfig string
+
+	cmd := &cobra.Command{
+		Use:     "apply",
+		Short:   "Applies policies on resources",
+		Example: fmt.Sprintf("To apply on a resource:\nkyverno apply /path/to/policy1 /path/to/policy2 --resource=/path/to/resource\n\nTo apply on a cluster\nkyverno apply /path/to/policy1 /path/to/policy2 --kubeConfig=/path/to/kubeConfig"),
+		RunE: func(cmd *cobra.Command, policyPaths []string) error {
+			if resourcePath == "" && kubeConfig == "" {
+				fmt.Println("Specify path to resource file or kube config")
+			}
+
+			var policies []*v1.ClusterPolicy
+			for _, policyPath := range policyPaths {
+				policy, err := getPolicy(policyPath)
+				if err != nil {
+					return err
+				}
+
+				policies = append(policies, policy)
+			}
+
+			resources, err := getResources(policies, kubeConfig, resourcePath)
+			if err != nil {
+				return err
+			}
+
+			for i, policy := range policies {
+				for j, resource := range resources {
+					if !(j == 0 && i == 0) {
+						fmt.Printf("\n\n=======================================================================\n")
+					}
+
+					err = applyPolicyOnResource(policy, resource)
+					if err != nil {
+						return err
+					}
+				}
+			}
+
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVar(&resourcePath, "resource", "", "path to resource file")
+	cmd.Flags().StringVar(&kubeConfig, "kubeConfig", "", "path to .kube/config file")
+	return cmd
+}
+
+func getResources(policies []*v1.ClusterPolicy, kubeConfig, resourcePath string) ([]*unstructured.Unstructured, error) {
+	var resources []*unstructured.Unstructured
+	var err error
+
+	if kubeConfig != "" {
+		var resourceTypesMap = make(map[string]bool)
+		var resourceTypes []string
+		for _, policy := range policies {
+			for _, rule := range policy.Spec.Rules {
+				for _, kind := range rule.MatchResources.Kinds {
+					resourceTypesMap[kind] = true
+				}
+			}
+		}
+
+		for kind := range resourceTypesMap {
+			resourceTypes = append(resourceTypes, kind)
+		}
+
+		resources, err = getResourcesOfTypeFromCluster(resourceTypes, kubeConfig)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if resourcePath != "" {
+		resource, err := getResource(resourcePath)
+		if err != nil {
+			return nil, err
+		}
+
+		resources = append(resources, resource)
+	}
+
+	return resources, nil
+}
+
+func getResourcesOfTypeFromCluster(resourceTypes []string, kubeConfig string) ([]*unstructured.Unstructured, error) {
+	var resources []*unstructured.Unstructured
+
+	clientConfig, err := config.CreateClientConfig(kubeConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	dClient, err := discovery.NewDiscoveryClientForConfig(clientConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	//var kindToListApi = map[string]string{
+	//	"pod": "/api/v1/pods",
+	//	"services": "/api/v1/services",
+	//}
+
+	for _, kind := range resourceTypes {
+		listObjectRaw, err := dClient.RESTClient().Get().RequestURI("/api/v1/" + strings.ToLower(kind) + "s").Do().Raw()
```

Comment:
Is there a better way to get all resources from a cluster? with this approach i will need to make list of api's to kind map from swagger.json. Please let me know @realshuting @shivdudhani @JimBugwadia